src/lib/Serialize.ts

```
@@ -0,0 +1,551 @@
+   * Converts raw data into a json format and returns it.
+   * @since 2.0.0
+   * @returns The formatted data.
+   */
+  public toJSON<Format extends Serialize.JSON | Serialize.JSON[] | Record<PropertyKey, Serialize.JSON>>(): Format {
+    return this.format().json as Format;
+  }
+
+  /**
+   * Converts json data into a raw format and returns it.
+   * @since 2.0.0
+   * @returns The raw data.
+   */
+  public toRaw(): unknown {
+    return this.parse().raw;
+  }
+
+  /**
+   * Converts raw data into a json format.
+   * @since 2.0.0
+   * @returns The {@link Serialize} class.
+   */
+  private format(): this {
+    if (Array.isArray(this.raw)) this.json = { type: Serialize.Type.Array, value: this.formatArray(this.raw) };
+    else if (typeof this.raw === 'bigint') this.json = { type: Serialize.Type.BigInt, value: this.raw.toString() };
+    else if (typeof this.raw === 'boolean') this.json = { type: Serialize.Type.Boolean, value: this.raw };
+    else if (this.raw instanceof Date) this.json = { type: Serialize.Type.Date, value: this.raw.toJSON() };
+    else if (this.raw instanceof Map) this.json = { type: Serialize.Type.Map, value: this.formatEntries(Array.from(this.raw)) };
+    else if (this.raw === null) this.json = { type: Serialize.Type.Null, value: null };
+    else if (typeof this.raw === 'number') this.json = { type: Serialize.Type.Number, value: this.raw };
+    else if (isObject(this.raw)) this.json = { type: Serialize.Type.Object, value: this.formatObject(this.raw) };
+    else if (this.raw instanceof RegExp) this.json = { type: Serialize.Type.RegExp, value: { source: this.raw.source, flags: this.raw.flags } };
+    else if (this.raw instanceof Set) this.json = { type: Serialize.Type.Set, value: this.formatArray(Array.from(this.raw)) };
+    else if (typeof this.raw === 'string') this.json = { type: Serialize.Type.String, value: this.raw };
+    else if (this.raw === undefined) this.json = { type: Serialize.Type.Undefined, value: 'undefined' };
+
```

What if the value isn't any of the recognized formats? We should throw an error rather than silently ignoring it.